stripe/src/main/java/com/stripe/android/model/KlarnaSourceParams.kt

```diff
@@ -0,0 +1,242 @@
+package com.stripe.android.model
+
+/**
+ * [Klarna Payments with Sources](https://stripe.com/docs/sources/klarna)
+ *
+ * Note:
+ * The optional fields [billingEmail], [billingPhone], [billingAddress], [billingFirstName],
+ * [billingLastName], and [billingDob] can be provided to skip Klarna's customer information form.
+ *
+ * If this information is missing, Klarna will prompt the customer for these values during checkout.
+ * Be careful with this option: If the provided information is invalid,
+ * Klarna may reject the transaction without giving the customer a chance to correct it.
+ */
+internal data class KlarnaSourceParams @JvmOverloads constructor(
+    /**
+     * The URL the customer should be redirected to after they have successfully verified the
+     * payment.
+     */
+    val purchaseCountry: String,
+
+    /**
+     * A list of [LineItem]. Klarna will present these on the confirmation page.
+     * The total amount charged will be a sum of the [LineItem.totalAmount] of each of these items.
+     */
+    val lineItems: List<LineItem>,
+
+    /**
+     * Required for customers located in the US. This determines whether Pay Later and/or Slice It
+     * is offered to a US customer.
+     */
+    val customPaymentMethods: Set<CustomPaymentMethods> = emptySet(),
+
+    /**
+     * An email address for the customer.
+     *
+     * Optional
+     */
+    val billingEmail: String? = null,
+
+    /**
+     * A phone number for the customer.
+     *
+     * Optional
+     */
+    val billingPhone: String? = null,
+
+    /**
+     * An [Address] for the customer. At a minimum, [Address.line1], [Address.postalCode],
+     * [Address.city], and [Address.country] must be provided.
+     *
+     * Optional
+     */
+    val billingAddress: Address? = null,
+
+    /**
+     * The customer's first name.
+     *
+     * Optional
+     */
+    val billingFirstName: String? = null,
+
+    /**
+     * The customer's last name.
+     *
+     * Optional
+     */
+    val billingLastName: String? = null,
+
+    /**
+     * The customer's date of birth.
+     * This will be used by Klarna for a credit check in some EU countries.
+     *
+     * Optional
+     */
+    val billingDob: DateOfBirth? = null,
+
+    /**
+     * See [Styling the Klarna Hosted Payment Page](https://stripe.com/docs/sources/klarna#styling-the-klarna-hosted-payment-page)
+     *
+     * You can customize the style of the Klarna hosted payment page by providing additional options
+     * when creating the source. Refer to the
+     * [Klarna SDK documentation](https://developers.klarna.com/en/us/kco-v3/hpp/4-customize/)
+     * for more information.
+     */
+    val pageOptions: PaymentPageOptions? = null
```

Comment:
Do these work? I couldn't get these to work.

Reply:
They don't right now. I'll follow up with the Klarna folks to figure out why.